ocaml-lsp-server/src/document.ml

```diff
@@ -83,10 +83,23 @@ let timer t = t.timer
 
 let source doc = Mpipeline.raw_source doc.pipeline
 
+let await task =
+  let open Fiber.O in
+  let () = Cancel.register (fun () -> Scheduler.cancel_task task) in
```

Comment:
I would expect `register` to be called for all of the requests, why only in `Document.with_pipeline`?

Reply:
There's nothing else that we can cancel. All of our cancellation points are essentially fixed at where we call `Ivar.read`, and we use ivar read for exactly two things: async & timers. Timers don't belong to any requests, so there's only async left.

We could make it possible to cancel formatting requests I suppose, but it seems more trouble than its worth.

Reply:
IMO we should try to make it general and work for all requests, but if you think that's too much trouble, let's wait until we need to register cancellations for other requests 🙂